providers/oidc.go

```diff
@@ -247,6 +255,11 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 		claims.UserID = fmt.Sprint(userID)
 	}
 
+	groups, ok := claims.rawClaims["groups"].([]string)
```

Comment:
I don't think the default `scopes` options has `groups` -- do you think that is something we potentially need (or documentation to add that if this option is used)?

Reply:
Makes sense will look at this.

Reply:
I tried this PR in our environment that is using  auth0. It seems to me that I hit some error here for the type assertion. The claims.rawClaims is map[string]interface{} and the groups cannot be extracted

Reply:
Interesting @lilida what do your claims look like?

Reply:
@JoelSpeed -- Is this a blocker for getting this merged until we get a better idea of what some of the major vendors do for the `groups` claim?  Or just start with assuming its standard lists inside of the groups claim and we can adapt as future uses try to enable it and have bugs?

Here's a discussion on Auth0's stance on it: https://community.auth0.com/t/group-scope-in-oidc/21317

@lilida -- Do you have a sample group claim from your Auth0 ID Token? Would help us figure out the types of corner cases we potentially need to handle.

Reply:
@NickMeves  I will try to get you some examples by the end of tomorrow.

Reply:
@NickMeves This is the payload we have. The   "https://xyz.io/groups" claim were added by our auth0 rules and I set it to be the --oidc-groups-claim

{
  "https://xyz.io/groups": [
    "dev|user",
    "dev|xyz",
    "abc|user",
    "abc|xyz",
    "efg|user",
    "efg|xyz",
    "demo-v2|user",
    "demo-v2|xyz"
  ],
  "https://xyz.io/metadata": {
    "roles": "user,xyz"
  },
  "https://xyz.io/email": "lida.li@xyz.com",
  "iss": "https://xyz.auth0.com/",
  "sub": "auth0|5f2846825f35ab0037b466ae",
  "aud": [
    "https://dev.xyz.io/",
    "https://xyz.auth0.com/userinfo"
  ],
  "iat": 1599673937,
  "exp": 1599760337,
  "azp": "70zBb3IEMHfVr3Sw55kwCn2jAR105PK5",
  "scope": "openid profile email"
}

Reply:
So it seems like the content of the claim is a standard list -- but the claim name is unconventional.  Did you set this option so it wasn't looking for the default `groups` as the key?

```
| `--oidc-groups-claim` | string | which claim contains the user groups | `"groups"` |
```

Reply:
Hmmm - the issue isn't jumping out at me... seems like we should be able to cast that groups list to a `[]string` 🤔 

This commit did go in yesterday to fix an issue related to Groups vs rawClaims groups conflicts: https://github.com/oauth2-proxy/oauth2-proxy/pull/616/commits/44bbd6d75f6cc7da13da3ab9fd56cbc4337be6ab

Reply:
@lilida @NickMeves can you try run this test with a new image built from my latest branch, looks like you ran things before I remove the unmarshalling of the `groups` and that may be causing some issues.

Looking at this with your claims and assuming you have `--oidc-groups-claim=https://xyz.io/groups` set, I would expect things to work.

Reply:
> Hmmm - the issue isn't jumping out at me... seems like we should be able to cast that groups list to a `[]string` 
> 
> This commit did go in yesterday to fix an issue related to Groups vs rawClaims groups conflicts: [44bbd6d](https://github.com/oauth2-proxy/oauth2-proxy/commit/44bbd6d75f6cc7da13da3ab9fd56cbc4337be6ab)

Yup you beat me by 20 seconds haha @NickMeves I suspect my fix may rectify this.

Reply:
Sure. I will test the latest branch https://github.com/stefansedich/oauth2-proxy/tree/group-claim and report back. I will be out for a few days so it may take some time.

Reply:
@lilida did you get a chance to test out the new build?

Reply:
@stefansedich I don't think I'm seeing any OIDC provider test cases that extract groups from a non-canonical claim customized by the user with `-oidc-groups-claim` (I didn't look very hard though, link me if I missed it).

I think we should add that scenario to the unit tests with a claim structure for groups similar to what @lilida posted since Auth0 is a decent sized IdP vendor.

If that passes unit tests, I think I'm comfortable merging this without hearing back on manual testing.

Reply:
@NickMeves ok great wrote a test and found a bug here, fixing this now!

Reply:
Ok @NickMeves can you check out the latest commit, there were two things that tripped me over here:

1. Even without the field definition groups was being unmarshalled, so now I explicitly clear the array if no claims are found based on the groups claim.
2. Using `claims.rawClaims[p.GroupsClaim].([]string)` does not actually work, it is actually an `[]interface{}` I now loop over each item casting to string safely, let me know if there is a better way to handle this.

Added a test case for using "groups" and "other_groups" and it looks to behave now, the hold that you suspected was actually there it was always unmarshaling "groups", my bad there should have written a test sooner :smile: